Makefile

```diff
@@ -39,60 +40,77 @@ ifeq ($(API_ONLY), 1)
   FLAGS += -Dapi_only
 endif
 
+# Development flags
+DEVFLAGS := --progress --stats --time
+
+# Output binary name
+BINARY_NAME := invidious
+BINARY_DEV := $(BINARY_NAME)-dev
 
 # -----------------------
 #  Main
 # -----------------------
 
-all: invidious
+.DEFAULT_GOAL := all
+
+all: $(BINARY_NAME)
 
 get-libs:
-	shards install --production
+	$(SHARDS_BIN) install --production
 
 # TODO: add support for ARM64 via cross-compilation
-invidious: get-libs
-	crystal build src/invidious.cr $(FLAGS) --progress --stats --error-trace
+$(BINARY_NAME): get-libs
+	$(CRYSTAL_BIN) build src/$(BINARY_NAME).cr $(FLAGS) --progress --stats --error-trace
 
-
-run: invidious
-	./invidious
+run: $(BINARY_NAME)
+	./$(BINARY_NAME)
 
 
 # -----------------------
 #  Development
 # -----------------------
 
-
 format:
-	crystal tool format
+	$(CRYSTAL_BIN) tool format --check
```

Comment:
Would prefer to have a separate arg for --check

Reply:
Separate arg or a different target altogether maybe?